src/plugins/vis_augmenter/common/augment_vis_saved_object_attributes.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectAttributes } from 'opensearch-dashboards/server';
+
+export interface AugmentVisSavedObjectAttributes extends SavedObjectAttributes {
+  id: string;
+  title: string;
+  description?: string;
+  originPlugin: string;
+  pluginResource: {
+    type: string;
+    id: string;
+  };
+  visLayerExpressionFn: {
+    type: string;
+    name: string;
+  };
+  version: number;
+  // Following fields are optional since they will get set/removed during the extraction/injection
+  // of the vis reference
+  visName?: string;
+  visId?: string;
+  visReference?: {
+    id: string;
+    name: string;
+  };
```

Comment:
Can we make a separate interface for these attributes? The convention for `SavedObjectAttributes` is to keep this interface identical to that of the saved object itself. If these properties are not going to be stored in the saved object, lets make a separate interface that extends this instead and use that. Also i dont see this being used anywhere, is it used in the AD plugin? if so can we just create the new interface with these properties there?

Reply:
These are used in the saved object actually - the `visName`, is populated to reference the `visualization` reference in the `references` array, and the `visId` and `visReference` are populated when pulling out the stored vis in `references`. This follows the same convention as used by visualization saved object `extractReferences/injectReferences` logic.

The visualization saved object uses the open-ended `SavedObjectAttributes`, and uses values there that aren't actually indexed in the saved object. I followed the same. From my understanding, the attributes can be values before/after the reference extraction/injection, based on the arguments to those functions themselves.

References:
visualization extract/inject references: [link](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/feature/feature-anywhere/src/plugins/visualizations/public/saved_visualizations/saved_visualization_references.ts)
augment-vis extract/inject references: [link](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/feature/feature-anywhere/src/plugins/vis_augmenter/public/saved_augment_vis/saved_augment_vis_references.ts)